### ISSUE TITLE
Mac OS Code generator adds back slashes before quotes

### ISSUE THREAD

**Issue by Nava2** [umple/umple#135]
Originally reported on Google Code with ID 235
Owned by @umple

---

**What steps will reproduce the problem?**

Generate a local version of UmpleOnline on a Mac and add an initialized string attribute,
e.g. xxx = "init:;

What is the expected output? 

Normal code

What do you see instead?

Generated code has = \"init\'",  In other words it adds back-slashes. This does not
occur when Umple is built on Linux.

I have heard anecdotal reports that this occurs when generating code on Eclipse using
the Mac. That needs confirming.

**Please use labels and text to provide additional information.**

---

Reported by @umple on 2011-10-06 20:47:56

**Comment by Nava2** [comment]
Looking into this.

---

Reported by @EmperorJordan on 2012-02-19 03:22:19

**Comment by Nava2** [comment]
Okay this is indeed very strange, and probably an important note in my search to solve
this:

When I compile the umple file containing:

// hello
class foo
{
  void testMethod()
  {
    // "inline comment inside a method."
    System.out.println("Wow backslashes");
  }
}

I get _different_ results depending on whether I generate the Java code in Umple online
(on my mac) or build using the command line (again on my mac).

Using UmpleOnline results in the backslashes before the quotes (in the comment and
inside the system.out).  Building on the command line results in NO backslashes.

This is interesting.

---

Reported by @EmperorJordan on 2012-02-19 03:31:54

**Comment by Nava2** [comment]
I have generated code, similar to the example I have mentioned in varying forms, and
cannot get the backslashes to appear in any way that the problem mentions in Eclipse.

So I think this is isolated to UmpleOnline and is probably something to do with javascript
or php, and logic within that formats things such that there is OS specific implementation
that does this.  I am very inexperienced with this stuff though so I may be running
in circles.

---

Reported by @EmperorJordan on 2012-02-19 03:56:34

**Comment by Nava2** [comment]
Going to leave this issue for now, as it has been deemed low priority for the moment.
 Only reproducible if you try this on a Mac and with a local version of UmpleOnline.

With that being said though I've debugged the javascript to the point where I know
the compiler is generating the code correctly, and it comes to the callback correctly
however where things are going wrong is likely in the file:

```
 umple_action.js
```

At the very bottom with the line:

```
 Ajax.sendRequest("scripts/compiler.php",callback,format("{0}&error={3}&umpleCode={1}&filename={2}",post,umpleCode,filename,errors));
```

Just before this line I printed out what the generated code was, and it was correct.
 Something with the php in the mac is messed (or something else) as its OS dependent.

---

Reported by @EmperorJordan on 2012-02-21 05:08:12
